### DISCUSSION TITLE
Should known hardware dependencies prohibit certain versions from being listed as available?

### DISCUSSION THREAD

**Discussion by stilett0** [microsoft/winget-pkgs#58559] [General] [answered]
I have an old computer that shipped with Windows Vista, but it has plenty of cores (for the time) and a reasonable amount of RAM, so it's been doing okay on Windows 10 x64.

The graphics card, however, I had upgraded to an NVidia GeForce GT 610, many years ago now.

NVidia no longer supports the NVidia GeForce GT 610 in their current drivers - and has not supported it for a couple of years now. I can understand that.

What I didn't see coming was their NVidia GeForce Experience utility. Starting with version 3.25.x,x, they've discontinued support for any NVidia chip prior to Maxwell.
https://www.nvidia.com/en-us/geforce/forums/geforce-experience/14/483833/geforce-experience-325-feedback-thread-released-27/
"Please note: The system requirements for GeForce Experience 3.25 has changed. The GeForce Experience 3.25 installer now requires Windows 10/Windows 11 running on a Maxwell based GeForce GPU or higher."

This was a fairly recent change.

I see that the winget-pkgs packages contain (currently) 3.25.1.27 and 3.25.0.84 in addition to several versions before.

I currently have (manually) installed GeForce Experience 3.24.0.126, which IS - so it seems - the final version compatible with the GeForce GT 610.

When running winget upgrade, I see the following in the report:
```
Name                    Id                             Version Available Source
-------------------------------------------------------------------------------
NVIDIA GeForce Experience Nvidia.GeForceExperience  3.24.0.126 3.25.1.27   winget
```

My question is, is it good for me to see 3.25.1.27 at all?

To be fair, if the upgrade is attempted with winget, 3.25.1.27 downloads, installs, and then fails. (mentioned here, I think: https://github.com/microsoft/winget-pkgs/pull/45689#pullrequestreview-878801856 )
And then the next time I run the winget upgrade, it is listed as available again...

This isn't the only utility I have like this, there's another system I have running the Intel Extreme Tuning Utility, and it seems that I cannot upgrade beyond 7.4.1.3 on that machine, as newer versions do not support that hardware.

(This has probably been discussed somewhere, but it didn't leap out at me, sorry.)

**Comment by OfficialEsco** [comment, marked as answer]
Your requirement is very niche but could probably be adapted under the Hardware requirement, however as GPU's can have different names i don't know how good it will work.. I think pinning would be a simple workaround, however that will not work for enterprises..

Currently we have exit code `3858759680` as a valid exit code to bypass the Hardware requirement of needing a Nvidia GPU so it can pass the validation which might be the same exit code you are getting with a unsupported GPU.
